docs(cloud): add Instill Credit documentation (#847)
instill-ai/instill.tech: docs/vdp/credit.en.mdx

---
title: "Instill Credit"
lang: "en-US"
draft: false
description: "Use Instill Credit in VDP"
---

The purpose of Credit is easing the adoption of Instill Cloud, minimizing the time required to build and set up a pipeline.
After setting up your account, you'll get **10,000 monthly credits for free**, which can be spent on the following actions:

> - Trigger your pipelines or any public pipeline on Instill Hub.

Author: @GeorgeWilliamStrong @xiaofei-du I struggled a bit wording the "1 credit per component" trigger cost as a feature or as something Credit unlocks. It is, rather, a limitation we set on the different pricing plans.

- Execute pre-configured AI Components without needing to create accounts or API keys on 3rd party services.

## Pipeline triggers

Each pipeline trigger will consume **1 credit per component** within the pipeline.

## AI components

Additionally, Instill AI allows out-of-the-box configuration for certain AI components.
When such components are created, if the selected task and model is supported, the default configuration will contain the credentials to start using the component.

<ZoomableImg
src="/docs-assets/vdp/credit-secret.png"
alt="Use Instill Credit to run DALL·E 3 image generation with pre-configured credentials"
width="400px"
/>

If you want to override the default configuration and bring your own key, you can update the **Connection** configuration in your component.

> Connection fields only take references to secrets in order to minimize the risk of leaking credentials when sharing your pipelines.

Author: @donch1989 @EiffelFly there are no references to secrets in the documentation, it'd be good to add a page / section on this 👀

---
title: "Instill Credit"
lang: "en-US"
draft: false
description: "Use Instill Credit in VDP"
---

> The purpose of Credit is easing the adoption of Instill Cloud, minimizing the time required to build and set up a pipeline.

Reviewer: ```suggestion
The purpose of Instill Credit is easing the adoption of Instill Cloud, minimizing the time required to build and set up a pipeline.
```

After setting up your account, you'll get **10,000 monthly credits for free**, which can be spent on the following actions:

> - Trigger your pipelines or any public pipeline on Instill Hub.

Reviewer: ```suggestion
- Trigger your own pipelines, or any public pipeline available on Instill Hub.
```

> - Execute pre-configured AI Components without needing to create accounts or API keys on 3rd party services.

Reviewer: ```suggestion
- Execute pre-configured AI components without needing to create accounts or API keys on 3rd party services.
```

Reviewer: This may change depending on what convention is agreed for referring to AI components. I have seen in the docs capitalisation and no capitalisation applied to components and I have raised this concern.

Author: Yeah I'm not aware of a convention so I didn't pay particular attention to this. I think we should only use capitalized `Component` when we refer to the entity (probably we don't have a use case for that in the documentation) or a particular one (e.g. OpenAI Component).

## Pipeline triggers

Each pipeline trigger will consume **1 credit per component** within the pipeline.

The **trigger** and **response** components don't consume any credit, as they are the pipeline's input and ouptut interfaces.

## AI components

Additionally, Instill AI allows out-of-the-box configuration for certain AI components.

> When such components are created, if the selected task and model is supported, the default configuration will contain the credentials to start using the component.

Reviewer: ```suggestion
When such components are created, if the selected task and model are supported, the default configuration will contain the credentials to start using the component.
```

| OpenAI       | DALL·E 3 Standard | 1024x1024            | 400                   |
| OpenAI       | DALL·E 3 Standard | 1024x1792, 1792x1024 | 800                   |
| OpenAI       | DALL·E 3 HD       | 1024x1024            | 800                   |
| OpenAI       | DALL·E 3 HD       | 1024x1792, 1792x1024 | 1,200                 |
| Stability AI | SDXL 1.0          |                      | 60                    |
| Stability AI | SD 1.6            |                      | 100                   |

#### Audio Recognition

| Vendor | Model   | Credit Cost per Second |
| :----- | :------ | :--------------------- |
| OpenAI | Whisper | 1                      |

#### Text to Speech

| Vendor | Model    | Credit Cost per 1M Characters |
| :----- | :------- | :---------------------------- |
| OpenAI | TTS 1    | 150                           |
| OpenAI | TTS 1 HD | 300                           |

#### Next additions

Check our [Roadmap](../roadmap) and subscribe to our [Newsletter](/newsletter/?utm_source=documentation&utm_medium=link) to keep up to date with the latest tasks, models and vendors that are supported by Instill Credit.

> We also plan to leverage Credit to ease data storage and model hosting, stay tuned!

Reviewer: ```suggestion
We also plan to leverage Instill Credit to ease data storage and model hosting, stay tuned!
```

Each pipeline trigger will consume **1 credit per component** within the pipeline.

> The **trigger** and **response** components don't consume any credit, as they are the pipeline's input and ouptut interfaces.

Author: cc @donch1989 can you confirm this?

Also, we never discussed the cost of the iterator component, how is this computed?

Reviewer: ```suggestion
This excludes the **trigger** and **response** components which don't consume any credit, as they are the pipeline's input and output interfaces.
```
